emerge-gitclone: do not touch commit if it is NoneType

dongsupark opened the pull request:
We should replace commit only if the commit is not of NoneType.

Otherwise it fails like that:
```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python2.7/emerge-gitclone", line 46, in <module>
    commit = commit.replace('refs/heads/', 'refs/remotes/origin/')
AttributeError: 'NoneType' object has no attribute 'replace'
```

margamanterola commented:
This is what gets the commit:
`commit = commits.get(repo.sync_uri.replace('//', '').replace('.git', '').split('/', 1)[-1])`

How can it be that this is empty? Is it cloning a repo with no commits?

margamanterola commented:
Ah, I've followed the code a bit more. I think I get it now.

margamanterola reviewed (approved):
LGTM

pothos commented on emerge-gitclone:
```diff
@@ -43,7 +43,8 @@ for repo in portage.db[eroot]['vartree'].settings.repositories:
 	# remote branch name, like "refs/remotes/origin/flatcar-build-x".
-	commit = commit.replace('refs/heads/', 'refs/remotes/origin/')
+	if commit:
+		commit = commit.replace('refs/heads/', 'refs/remotes/origin/')
```
Can we add an else case here where we print a warning?
```
else:
    print("Warning: No revision found for " + repo.sync_uri)
```

dongsupark replied:
Done

pothos replied:
Thanks

Commit:
emerge-gitclone: do not touch commit if it is NoneType

We should replace commit only if the commit is not of NoneType.

Otherwise it fails like that:
```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python2.7/emerge-gitclone", line 46, in <module>
    commit = commit.replace('refs/heads/', 'refs/remotes/origin/')
AttributeError: 'NoneType' object has no attribute 'replace'
```
